File: rpg-project/backend/src/models/Item.js
import { BOOLEAN, INTEGER, STRING } from "sequelize";
import db from "../config/db.js";
import { EQUIP_TYPE, ITEM_TYPE } from "../utils/itemType.js";
import Inventory from "./Inventory.js";
import User from "./User.js";

const Item = db.define(
    "item",
    {
        vnum: {
            type: INTEGER,
            unique: true,
            allowNull: false,
        },
        item_name: {
            type: STRING,
            allowNull: false,
        },
        rarity: {
            type: INTEGER,
            allowNull: false,
            defaultValue: 0,
        },
        type: {
            type: INTEGER,
            allowNull: false,
            defaultValue: ITEM_TYPE.MATERIAL,
        },
        subtype: {
            type: INTEGER,
            allowNull: false,
            defaultValue: EQUIP_TYPE.NONE,
        },
        price: {
            type: INTEGER(9),
            allowNull: false,
            defaultValue: 0,
        },
    },
    { timestamps: false }
);

export default Item;

File: rpg-project/RPGApi/RPGApi/Item.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RPGApi
{
    public class Item
    {
        public long vnum { get; set; }
        public string item_name { get; set; }
        public int rarity { get; set; }
        public int type { get; set; }
        public int subtype { get; set; }
        public int price { get; set; }

        public Item(long vnum, string item_name, int rarity, int type, int subtype, int price)
        {
            this.vnum = vnum;
            this.item_name = item_name;
            this.rarity = rarity;
            this.type = type;
            this.subtype = subtype;
            this.price = price;
        }
    }
}
